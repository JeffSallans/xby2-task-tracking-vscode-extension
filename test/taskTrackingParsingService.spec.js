/* global suite, test */

//
// Note: This example test is leveraging the Mocha test framework.
// Please refer to their documentation on https://mochajs.org/ for help.
//

// The module 'assert' provides assertion methods from node
const fs = require('fs');
const moment = require('moment');

// You can import and use all API from the 'vscode' module
// as well as import your extension to test it
const taskTrackingService = require('../taskTrackingParsingService');
const getTasksMonthView = fs.readFileSync('./test/mock-requests/get-tasks-month-view.html').toString();
const getTasksMonthOctoberView = fs.readFileSync('./test/mock-requests/get-tasks-month-view-oct.html').toString();
const getTaskDetails = fs.readFileSync('./test/mock-requests/get-task-details.html').toString();

// Defines a Mocha test suite to group tests of similar kind together
describe("Task Tracking Parsing HTML Service", function() {

    // Defines a Mocha unit test
    it("parseOutActivityIdListGivenDate to work for mock html", function() {
        const activityIds = taskTrackingService.parseOutActivityIdListGivenDate(getTasksMonthView, moment('08-02-2017', 'MM-DD-YYYY'));
        expect(activityIds).toMatchSnapshot();
    });

    it("parseOutActivityDetails to work for mock html", function() {
        const task = taskTrackingService.parseOutActivityDetails(getTaskDetails, 229268);
        expect(task).toMatchSnapshot();
    });

    it("parseOutTasksOfTheWeek to work for mock html", function() {
        const tasks = taskTrackingService.parseOutTasksOfTheWeek(getTasksMonthView, moment('08-02-2017', 'MM-DD-YYYY'));
        expect(tasks).toMatchSnapshot();
    });

    it("parseOutTasksOfTheWeek to work for october example html", function() {
        const tasks = taskTrackingService.parseOutTasksOfTheWeek(getTasksMonthOctoberView, moment('10-05-2017', 'MM-DD-YYYY'));
        expect(tasks).toMatchSnapshot();
    });

    it("parseOutTasksOfTheMonth to work for mock html", function() {
        const tasks = taskTrackingService.parseOutTasksOfTheMonth(getTasksMonthView, moment('08-02-2017', 'MM-DD-YYYY'));
        expect(tasks).toMatchSnapshot();
    });
});