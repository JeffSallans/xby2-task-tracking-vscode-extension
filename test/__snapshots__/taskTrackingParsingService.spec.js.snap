// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Tracking Parsing HTML Service parseOutActivityDetails to work for mock html 1`] = `
Object {
  "activityId": 229268,
  "clientId": 48,
  "clientName": "Auto-Owners Insurance",
  "date": "2017-08-02T04:00:00.000Z",
  "description": "Hardware issue, need to replace laptop<br>Technical QA documentation<br>React performance Prototype",
  "hours": 8,
  "isBillable": true,
  "minutes": 0,
  "projectId": 552,
  "projectName": "UWÂ Modernization",
  "taskId": 4435,
  "taskName": "Implementation",
}
`;

exports[`Task Tracking Parsing HTML Service parseOutActivityIdListGivenDate to work for mock html 1`] = `
Array [
  "229268",
]
`;

exports[`Task Tracking Parsing HTML Service parseOutTasksOfTheMonth to work for mock html 1`] = `
Array [
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-01T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-02T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-03T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-04T04:00:00.000Z",
    "description": "",
    "hours": 4,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
]
`;

exports[`Task Tracking Parsing HTML Service parseOutTasksOfTheWeek to work for mock html 1`] = `
Array [
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-07-31T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-01T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-02T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-03T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-04T04:00:00.000Z",
    "description": "",
    "hours": 4,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
]
`;
